src/test/ui/nll/user-annotations/dump-fn-method.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Unit test for the "user substitutions" that are annotated on each
+// node.
+
+// compile-flags:-Zverbose
+
+#![feature(nll)]
+#![feature(rustc_attrs)]
+
+trait Bazoom<T> {
```

Comment:
You might want to just comment here that the names `T` and `U` are significant in the sense that the comments for each test reference them to identify how the user-written type is yielding the particular components of the user substs that is printed by the test.

(or at least, when I first started reading the comments, I asked "wait, what is `T`? And `U`???" Because I hadn't even registered the names in my head when I read the `trait` and `impl` definitions here.